pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -0,0 +1,339 @@
+		snapshotSchedule.Status.Items[policyType] = make([]*stork_api.ScheduledVolumeSnapshotStatus, 0)
+	}
+	snapshotSchedule.Status.Items[policyType] = append(snapshotSchedule.Status.Items[policyType],
+		&stork_api.ScheduledVolumeSnapshotStatus{
+			Name:              snapshotName,
+			CreationTimestamp: meta.Now(),
+			Status:            snapv1.VolumeSnapshotConditionPending,
+		})
+	err := sdk.Update(snapshotSchedule)
+	if err != nil {
+		return err
+	}
+
+	snapshot := &snapv1.VolumeSnapshot{
+		Metadata: meta.ObjectMeta{
+			Name:        snapshotName,
+			Namespace:   snapshotSchedule.Namespace,
+			Annotations: snapshotSchedule.Annotations,
+			Labels:      snapshotSchedule.Labels,
+		},
+		Spec: snapshotSchedule.Spec.Template.Spec,
+	}
+	log.VolumeSnapshotScheduleLog(snapshotSchedule).Infof("Starting snapshot %v", snapshotName)
+	// If reclaim policy is set to Delete, this will delete the snapshots
+	// created by this snapshotschedule when the schedule object is deleted
+	if snapshotSchedule.Spec.ReclaimPolicy == stork_api.ReclaimPolicyDelete {
+		snapshot.Metadata.OwnerReferences = []meta.OwnerReference{
+			{
+				Name:       snapshotSchedule.Name,
+				UID:        snapshotSchedule.UID,
+				Kind:       snapshotSchedule.GetObjectKind().GroupVersionKind().Kind,
+				APIVersion: snapshotSchedule.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+			},
+		}
+	}
+	_, err = k8s.Instance().CreateSnapshot(snapshot)
+	return err
+}
+
+func (s *SnapshotScheduleController) pruneVolumeSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) error {
+	for policyType, policyVolumeSnapshot := range snapshotSchedule.Status.Items {
+		numVolumeSnapshots := len(policyVolumeSnapshot)
+		deleteBefore := 0
+		retainNum, err := schedule.GetRetain(snapshotSchedule.Spec.SchedulePolicyName, policyType)
+		if err != nil {
+			return err
+		}
+		numReady := 0
+
+		// Keep up to retainNum successful snapshot statuses and all failed snapshots
+		// until there is a successful one
+		if numVolumeSnapshots > int(retainNum) {
+			// Start from the end and find the retainNum successful snapshots
+			for i := range policyVolumeSnapshot {
+				if policyVolumeSnapshot[(numVolumeSnapshots-1-i)].Status == snapv1.VolumeSnapshotConditionReady {
+					numReady++
+					if numReady > int(retainNum) {
+						deleteBefore = numVolumeSnapshots - i
+						break
+					}
+				}
+			}
+			if numReady > int(retainNum) {
+				for i := 0; i < deleteBefore; i++ {
+					err := k8s.Instance().DeleteSnapshot(policyVolumeSnapshot[i].Name, snapshotSchedule.Namespace)
+					if err != nil {
+						log.VolumeSnapshotScheduleLog(snapshotSchedule).Warnf("Error deleting %v: %v", policyVolumeSnapshot[i].Name, err)
+					}
+				}
+			}
+			snapshotSchedule.Status.Items[policyType] = policyVolumeSnapshot[deleteBefore:]
+		}
+	}
+	return sdk.Update(snapshotSchedule)
+}
+
+func (s *SnapshotScheduleController) deleteVolumeSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) error {
```

Comment:
Is this being used anywhere?

Reply:
It was being used to clean up snapshots on deleting the schedule. But I added the owner ref in the snapshots which takes care of that now. Removed this.